src/R/Core/Impl/AST/Operators/Associativity.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.R.Core.AST.Operators {
+    // https://en.wikipedia.org/wiki/Operator_associativity
+    /// <summary>
+    /// Associativity. Consider the expression a~b~c. If the operator 
+    /// ~ has left associativity, this expression would be interpreted as (a~b)~c.
+    /// If the operator has right associativity, the expression would be
+    /// interpreted as a~(b~c).
+    /// </summary>
+    public enum Associativity {
```

Comment:
Yes, I think we should check syntax deeper (there is a separate task for async syntax check with better analysis and separate error/warning messages). I.e. we hopefully should be detecting what will fail at runtime but not necessarily allow everything that runtime allows.